WhoIsThat/WhoIsThat/ViewModels/LoginViewModel.cs

```diff
@@ -1,25 +1,115 @@
-﻿using System;
+﻿using Plugin.Media.Abstractions;
+using Plugin.Permissions;
+using Plugin.Permissions.Abstractions;
+using System;
 using System.Collections.Generic;
 using System.ComponentModel;
+using System.Runtime.CompilerServices;
 using System.Text;
 using System.Windows.Input;
+using WhoIsThat.Connections;
+using WhoIsThat.Handlers;
+using WhoIsThat.Handlers.Utils;
+using WhoIsThat.Models;
 using Xamarin.Forms;
 
 namespace WhoIsThat.ViewModels
 {
-    public class LoginViewModel
+    public class LoginViewModel : INotifyPropertyChanged
     {
         public ICommand NavigateHomePageCommand { get; private set; }
+        public ICommand TakePhotoCommand { get; set; }
+        private RestService _restService;
+        public LoginViewModel()
+        {
+            NavigateHomePageCommand = new Command(SavePerson);
+            TakePhotoCommand = new Command(TakePhoto);
+            PersonObject = new ImageObject();
+            _restService = new RestService();
+        }
+        
+
         public INavigation Navigation { get; set; }
 
-        public LoginViewModel()
+        public event PropertyChangedEventHandler PropertyChanged;
+
+                
+        private MediaFile takenPhoto { get; set; }
+        private ImageObject personObject;
+
+        public ImageObject PersonObject
+        {
+            get
+            {
+                return personObject;
+            }
+            set
+            {
+                personObject = value;
+                OnPropertyChanged();
+            }
+        }
+
+        public async void TakePhoto()
         {
-            NavigateHomePageCommand = new Command(NavigateToHomePage);
+            //Checking for camera permissions
+            bool cameraPermission = await PermissionHandler.CheckForCameraPermission();
+            if (!cameraPermission)
+                await CrossPermissions.Current.RequestPermissionsAsync(Permission.Camera);
+
+            //Checking for storage permissions
+            bool storagePermission = await PermissionHandler.CheckForCameraPermission();
+            if (!storagePermission)
+                await CrossPermissions.Current.RequestPermissionsAsync(Permission.Storage);
+
+            //Taking photo and storing it in MediaFile variable 'takenPhoto'
+
+            takenPhoto = await TakingPhotoHandler.TakePhoto();
+
+            //Save taken photo to Azure cloud for recognition, later on it is deleted
+
+            //await CloudStorageService.SaveBlockBlob(takenPhoto);
+        }
+
+        public async void SavePerson()
+        { 
+            //Checks if all required fields are filled
+
+            /*if(PersonObject.PersonFirstName == null || PersonObject.PersonLastName == null || PersonObject.DescriptiveSentence == null || takenPhoto == null)
+            {
+                return;
+            }*/
+
+            //If so, stores his information in the db
+            PersonObject.Id = 1;
```

Comment:
ID is automatically generated in backend, so you don't need to specify it here

Reply:
true, i used this to see, if the app remembers the id stored in application properties. It's removed now